include/rtc/rtp.hpp

```diff
@@ -0,0 +1,506 @@
+        // "length" in packet is one less than the number of 32 bit words in the packet.
+        size_t length = (sizeWithReportBlocks(reportCount) / 4) - 1;
+        header.prepareHeader(201, reportCount, uint16_t(length));
+        this->_senderSSRC = htonl(senderSSRC);
+    }
+
+    inline static size_t sizeWithReportBlocks(uint8_t reportCount) {
+        return sizeof(header) + 4 + size_t(reportCount) * sizeof(RTCP_ReportBlock);
+    }
+
+    inline bool isReceiverReport() {
+        return header.payloadType() == 201;
+    }
+
+    inline void log() const {
+        header.log();
+        PLOG_VERBOSE << "RTCP RR: "
+                   << " SSRC=" << ntohl(_senderSSRC);
+
+        for (unsigned i = 0; i < unsigned(header.reportCount()); i++) {
+            getReportBlock(i)->log();
+        }
+    }
+};
+
+
+struct RTCP_REMB {
+    RTCP_FB_HEADER header;
+
+    /*! \brief Unique identifier ('R' 'E' 'M' 'B') */
+    char id[4];
+
+    /*! \brief Num SSRC, Br Exp, Br Mantissa (bit mask) */
+    uint32_t bitrate;
+
+    SSRC ssrc[1];
+
+    [[nodiscard]] unsigned int getSize() const {
+        // "length" in packet is one less than the number of 32 bit words in the packet.
+        return sizeof(uint32_t) * (1 + header.header.length());
+    }
+
+    void preparePacket(SSRC senderSSRC, unsigned int numSSRC, unsigned int bitrate) {
+
+        // Report Count becomes the format here.
+        header.header.prepareHeader(206, 15, 0);
+
+        // Always zero.
+        header.setMediaSourceSSRC(0);
+
+        header.setPacketSenderSSRC(senderSSRC);
+
+        id[0] = 'R';
+        id[1] = 'E';
+        id[2] = 'M';
+        id[3] = 'B';
+
+        setBitrate(numSSRC, bitrate);
+    }
+
+    void setBitrate(unsigned int numSSRC, unsigned int bitrate) {
+        unsigned int exp = 0;
+        while (bitrate > pow(2, 18) - 1) {
+            exp++;
+            bitrate /= 2;
+        }
+
+        // "length" in packet is one less than the number of 32 bit words in the packet.
+        header.header.setLength((offsetof(RTCP_REMB, ssrc) / sizeof(uint32_t)) - 1 + numSSRC);
+
+        this->bitrate = htonl(
+                (numSSRC << (32u - 8u)) | (exp << (32u - 8u - 6u)) | bitrate
+        );
+    }
+
+    void setSsrc(int iterator, SSRC newSssrc){
+        ssrc[iterator] = htonl(newSssrc);
+    }
+    
+    size_t static inline sizeWithSSRCs(int count) {
+        return sizeof(RTCP_REMB) + (count-1)*sizeof(SSRC);
+    }
+};
+
+
+struct RTCP_PLI {
+    RTCP_FB_HEADER header;
+
+    void preparePacket(SSRC messageSSRC) {
+        header.header.prepareHeader(206, 1, 2);
+        header.setPacketSenderSSRC(messageSSRC);
+        header.setMediaSourceSSRC(messageSSRC);
+    }
+
+    void print() {
+        header.log();
+    }
+
+    [[nodiscard]] static unsigned int size() {
+        return sizeof(RTCP_FB_HEADER);
+    }
+};
+
+struct RTCP_FIR_PART {
+    uint32_t ssrc;
+#if __BYTE_ORDER == __BIG_ENDIAN
+    uint32_t seqNo: 8;
+    uint32_t: 24;
```

Comment:
The field sizes are actually just a suggestion for the compiler, it's not required to pack the data. It is safer to rely on the size of fields, which are guaranteed without padding since pack is set to 1 byte at the top of the file. Also, it's cleaner as you don't have to mess with byte ordering.

Here, you could use:
```
uint8_t _seqNo;
uint8_t _dummy1;
uint16_t _dummy2;
```